lib/matplotlib/table.py

```diff
@@ -472,6 +524,9 @@ def table(ax,
 
     Thanks to John Gill for providing the class and table.
     """
+    if cellType is not None:
+        assert cellType in ('default', 'scicell')
```

Comment:
We just (well in the process of #3060 needs one more read through before it is merged) converting all of the asserts into raises.  Please raise a `ValueError` here.

The logic behind this is that if it is really a problem that will cause the code to fail, then it should not be possible to avoid the checks (which you can do with asserts via interpreter flags).